### PR TITLE
[boost] update to 1.77.0

### LINKED ISSUE #8753: [boost-lexical-cast] Depends on boost.math?
It seems that `boost-lexical-cast` depends on `boost-math`:
https://github.com/boostorg/lexical_cast/blob/87ce72192836787ff7bc8628abf0f0a68d56001d/include/boost/lexical_cast/detail/inf_nan.hpp#L34-L35
But `boost-math` is not in its `Build-Depends` list.

I tried to add `boost-math` to `boost-lexical-cast`'s `Build-Depends`:
https://github.com/microsoft/vcpkg/pull/8752/commits/d1ce9dc03b76e22f25c49811259bb72c88c18def (https://github.com/microsoft/vcpkg/pull/8752)
and the CI system reports an error:
https://dev.azure.com/vcpkg/public/_build/results?buildId=17136

### LINKED ISSUE #10609: boost --triplet x64-uwp build failure
**Host Environment**
- OS: Windows 1909 (OS Build 18363.628)
- Compiler: VS2019 Version 16.3.9

**To Reproduce**
`.\vcpkg.exe install boost --triplet x64-uwp`

**Failure logs** 
Error: Building package boost-fiber:x64-uwp failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: boost-fiber:x64-uwp
  Vcpkg version: 2020.02.04-nohash-external

**Additional context**

After checking, it seems following packages were built despite point of failure in the above:

boost-foreach_x64-uwp
boost-format_x64-uwp
boost-function-types_x64-uwp
boost-functional_x64-uwp
boost-function_x64-uwp
boost-fusion_x64-uwp
boost-graph_x64-uwp
boost-hof_x64-uwp
boost-integer_x64-uwp
boost-interprocess_x64-uwp
boost-interval_x64-uwp
boost-intrusive_x64-uwp
boost-io_x64-uwp
boost-iterator_x64-uwp
boost-lambda_x64-uwp
boost-lexical-cast_x64-uwp
boost-logic_x64-uwp
boost-math_x64-uwp
boost-modular-build-helper_x64-uwp
boost-move_x64-uwp
boost-mp11_x64-uwp
boost-mpl_x64-uwp
boost-multi-array_x64-uwp
boost-multi-index_x64-uwp
boost-multiprecision_x64-uwp
boost-numeric-conversion_x64-uwp
boost-odeint_x64-uwp
boost-optional_x64-uwp
boost-parameter_x64-uwp
boost-phoenix_x64-uwp
boost-pool_x64-uwp
boost-predef_x64-uwp
boost-preprocessor_x64-uwp
boost-property-map_x64-uwp
boost-property-tree_x64-uwp
boost-proto_x64-uwp
boost-ptr-container_x64-uwp
boost-random_x64-uwp
boost-range_x64-uwp
boost-rational_x64-uwp
boost-ratio_x64-uwp
boost-regex_x64-uwp
boost-serialization_x64-uwp
boost-signals2_x64-uwp
boost-smart-ptr_x64-uwp
boost-spirit_x64-uwp
boost-static-assert_x64-uwp
boost-system_x64-uwp
boost-thread_x64-uwp
boost-throw-exception_x64-uwp
boost-tokenizer_x64-uwp
boost-tti_x64-uwp
boost-tuple_x64-uwp
boost-type-index_x64-uwp
boost-type-traits_x64-uwp
boost-typeof_x64-uwp
boost-ublas_x64-uwp
boost-units_x64-uwp
boost-unordered_x64-uwp
boost-utility_x64-uwp
boost-uuid_x64-uwp
boost-variant_x64-uwp
boost-vcpkg-helpers_x64-uwp
boost-vmd_x64-uwp
boost-winapi_x64-uwp
boost-xpressive_x64-uwp

Lastly, regression testing should be in place for libs that formerly worked so that new builds are not introduced until all supported platforms are in working order.

### LINKED ISSUE #16580: [boost-iostreams] build fails to include zstd
**Host Environment**
- OS: Ubuntu 20.04 (works on Windows)
- Compiler: gcc

**To Reproduce**
`./vcpkg install boost-iostreams`

**Failure** 
The port builds fine, but only zlib and bzip2 are considered during compilation, even though for ex zstd is installed already with vcpkg and it seems to be listed as dependency in the port script. When trying to use some functionality that depends on zstd I get many linking errors like this:
```
[build] [ 81%] Linking CXX executable ../../bin/Data
[build] /usr/bin/ld: ../../lib/libMVS.a(Scene.cpp.o): in function `boost::iostreams::detail::zstd_decompressor_impl<std::allocator<char> >::~zstd_decompressor_impl()':
[build] /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:310: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:310: undefined reference to `boost::iostreams::detail::zstd_base::~zstd_base()'
[build] /usr/bin/ld: ../../lib/libMVS.a(Scene.cpp.o): in function `boost::iostreams::detail::zstd_compressor_impl<std::allocator<char> >::~zstd_compressor_impl()':
[build] /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:286: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:286: undefined reference to `boost::iostreams::detail::zstd_base::~zstd_base()'
[build] /usr/bin/ld: ../../lib/libMVS.a(Scene.cpp.o): in function `boost::iostreams::detail::zstd_decompressor_impl<std::allocator<char> >::close()':
[build] /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:331: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
[build] /usr/bin/ld: ../../lib/libMVS.a(Scene.cpp.o): in function `boost::iostreams::detail::zstd_decompressor_impl<std::allocator<char> >::filter(char const*&, char const*, char*&, char*, bool)':
[build] /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:324: undefined reference to `boost::iostreams::detail::zstd_base::before(char const*&, char const*, char*&, char*)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:325: undefined reference to `boost::iostreams::zstd::finish'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:325: undefined reference to `boost::iostreams::zstd::run'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:325: undefined reference to `boost::iostreams::detail::zstd_base::inflate(int)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:326: undefined reference to `boost::iostreams::detail::zstd_base::after(char const*&, char*&, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:327: undefined reference to `boost::iostreams::zstd::stream_end'
[build] /usr/bin/ld: ../../lib/libMVS.a(Scene.cpp.o): in function `boost::iostreams::detail::zstd_compressor_impl<std::allocator<char> >::filter(char const*&, char const*, char*&, char*, bool)':
[build] /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:293: undefined reference to `boost::iostreams::detail::zstd_base::before(char const*&, char const*, char*&, char*)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:294: undefined reference to `boost::iostreams::zstd::finish'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:294: undefined reference to `boost::iostreams::zstd::run'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:294: undefined reference to `boost::iostreams::detail::zstd_base::deflate(int)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:295: undefined reference to `boost::iostreams::detail::zstd_base::after(char const*&, char*&, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:296: undefined reference to `boost::iostreams::zstd::stream_end'
[build] /usr/bin/ld: ../../lib/libMVS.a(Scene.cpp.o): in function `boost::iostreams::detail::zstd_compressor_impl<std::allocator<char> >::close()':
[build] /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:300: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:300: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:300: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
[build] /usr/bin/ld: /home/wii/Pro/vcpkg/installed/x64-linux/include/boost/iostreams/filter/zstd.hpp:300: undefined reference to `boost::iostreams::detail::zstd_base::reset(bool, bool)'
```

### LINKED ISSUE #17312: [boost] Failed when build external libtorrent that uses boost library.
**Describe the bug**
```
D:\vcpkg\installed\arm64-windows-static\include\boost/multiprecision/cpp_int/intel_intrinsics.hpp(103,11): error C3861:
 '_addcarry_u32': identifier not found
```

**Environment**
- OS: windows x64 cross compile to arm64
- Compiler: visual studio 2019

**To Reproduce**
1. ./vcpkg install --triplet=arm64-windows boost openssl
2. build latest libtorrent manually

**Expected behavior**
build success

**Failure logs** 
```
D:\libtorrent-rasterbar-2.0.3\build>vcpkg --triplet=arm64-windows-static install boost openssl
...................................
D:\libtorrent-rasterbar-2.0.3\build>cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE="D:\vcpkg\scripts\buildsystems\vcpkg.cmake" -DVCPKG_TARGET_TRIPLET="arm64-windows-static" -DBUILD_SHARED_LIBS=OFF -Dstatic_runtime=ON ..
-- The C compiler identification is MSVC 19.28.29913.0
-- The CXX compiler identification is MSVC 19.28.29913.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: D:/MicrosoftVisualStudio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/arm64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: D:/MicrosoftVisualStudio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/arm64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB - Success
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Found OpenSSL: D:/vcpkg/installed/arm64-windows-static/lib/libcrypto.lib (found version "1.1.1j")
-- Found Boost: D:/vcpkg/installed/arm64-windows-static/include (found version "1.75.0")
-- The following features have been enabled:

 * static_runtime, build libtorrent with static runtime
 * dht, enable support for Mainline DHT
 * deprecated-functions, enable deprecated functions for backwards compatibility
 * encryption, Enables encryption in libtorrent
 * exceptions, build with exception support
 * extensions, Enables protocol extensions
 * i2p, build with I2P support
 * logging, build with logging
 * mutable-torrents, Enables mutable torrent support
 * streaming, Enables support for piece deadline

-- The following RECOMMENDED packages have been found:

 * OpenSSL, Full-strength general purpose cryptography library, <https://www.openssl.org/>
   Provides HTTPS support to libtorrent

-- The following REQUIRED packages have been found:

 * Threads
 * Boost

-- The following features have been disabled:

 * BUILD_SHARED_LIBS, build libtorrent as a shared library
 * build_tests, build tests
 * build_examples, build examples
 * build_tools, build tools
 * python-bindings, build python bindings
 * python-egg-info, generate python egg info
 * python-install-system-dir, Install python bindings to the system installation directory rather than the CMake installation prefix
 * gnutls, build using GnuTLS instead of OpenSSL

-- Configuring done
-- Generating done
-- Build files have been written to: D:/libtorrent-rasterbar-2.0.3/build

D:\libtorrent-rasterbar-2.0.3\build>cmake --build .
[45/164] Building CXX object CMakeFiles\torrent-rasterbar.dir\src\peer_connection.cpp.obj
FAILED: CMakeFiles/torrent-rasterbar.dir/src/peer_connection.cpp.obj
D:\MicrosoftVisualStudio\2019\Community\VC\Tools\MSVC\14.28.29910\bin\Hostx64\arm64\cl.exe  /nologo /TP -DBOOST_ALL_NO_LIB -DBOOST_ASIO_ENABLE_CANCELIO -DBOOST_ASIO_HAS_STD_CHRONO -DBOOST_ASIO_NO_DEPRECATED -DBOOST_EXCEPTION_DISABLE -DOPENSSL_NO_SSL2 -DTORRENT_BUILDING_LIBRARY -DTORRENT_SSL_PEERS -DTORRENT_USE_LIBCRYPTO -DTORRENT_USE_OPENSSL -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE -D_FILE_OFFSET_BITS=64 -D_SCL_SECURE_NO_DEPRECATE -D_WIN32_WINNT=0x0600 -I..\include -I..\deps\try_signal -ID:\vcpkg\installed\arm64-windows-static\include /DWIN32 /D_WINDOWS /GR /EHsc /O2 /Ob2 /DNDEBUG -MT /Zc:__cplusplus /W4 /wd4251 /wd4275 /wd4373 /wd4268 /wd4503 /Zc:wchar_t /Zc:forScope /MP /bigobj /EHsc /showIncludes /FoCMakeFiles\torrent-rasterbar.dir\src\peer_connection.cpp.obj /FdCMakeFiles\torrent-rasterbar.dir\torrent-rasterbar.pdb /FS -c ..\src\peer_connection.cpp
D:\vcpkg\installed\arm64-windows-static\include\boost/multiprecision/cpp_int/intel_intrinsics.hpp(103): error C3861: '_addcarry_u32': identifier not found
D:\vcpkg\installed\arm64-windows-static\include\boost/multiprecision/cpp_int/intel_intrinsics.hpp(108): error C3861: '_subborrow_u32': identifier not found
[52/164] Building CXX object CMakeFiles\torrent-rasterbar.dir\src\http_connection.cpp.obj
ninja: build stopped: subcommand failed.
```
**Additional context**

### LINKED ISSUE #17882: [boost] Fix dependency generation by vcpkg/scripts/boost/generate-ports.ps1
**Is your feature request related to a problem? Please describe.**
Boost modular ports generated by a script in vcpkg do not report the correct list of dependencies in "vcpkg.json" files. Observed while working on a new port relying on boost dynamic-bitset #17848 

Currently, the generator script searches in the include and source directories for strings to find included headers files and deduce the dependency.

```powershell
        if ($IsWindows)
        {
            $groups = $(
                findstr /si /C:"include <boost/" include/*
                findstr /si /C:"include <boost/" src/*
            ) | % { $_ -replace "^[^:]*:","" }
        }
        else
        {
            $groups = $(
                grep -irhs "include <boost/" include src
            )
        }
```
However, it searches only headers starting as `<boost/` and omits the one starting as `"boost/`

I did not manage to run the script successfully for testing it, so I leave this here as an issue.

**Proposed solution**
Search also for header file starting with `"boost/`

### LINKED ISSUE #19506: [boost-multiprecision] missing dependency
**Describe the bug**
Tried to compile simple program with boost-multiprecision and compiler complains about missing boost/math/tools/complex.hpp file

**Environment**
- OS: Windows 10
- Compiler: MSVC 19

**To Reproduce**
1. `vcpkg install boost-multiprecision:x64-windows`
2. Add boost to CMakeLists: find_package(Boost REQUIRED); target_link_libraries(Test PRIVATE Boost::boost)
3. include boost multiprecision header in a simple hello world program
4. Compile: fatal error C1083: Cannot open include file: 'boost/math/tools/complex.hpp' 

**Expected behavior**
compile successfully

**Additional context**
Everything works after manually installing boost-math
`vcpkg install boost-math:x64-windows`

### LINKED ISSUE #19555: [Boost] update to 1.77.0
Library name: **Boost**

New version number: **1.77.0**

Other information that may be useful (release notes, etc...)
[Boost Version History](https://www.boost.org/users/history/)
[Release Notes](https://www.boost.org/users/history/version_1_77_0.html)

https://github.com/boostorg/boost/releases/latest

### PR DESCRIPTION
**Pull request by yurybura**
Updates Boost to 1.77.0

Fixes #19555
Fixes #19506
Fixes #17882
Fixes #17312
Fixes #16580
Fixes #10609
Fixes #8753

Supersedes #19297
Supersedes #19514

### CODE REVIEW

**Comment by BillyONeal** [comment]
Pushed merge with master because existing VM fleet is being shut down.

File: scripts/boost/generate-ports.ps1 [hunk 1]
```diff
@@ -369,31 +371,9 @@ foreach ($library in $libraries)
 
         $deps = @($deps | ? {
             # Boost contains cycles, so remove a few dependencies to break the loop.
```
Comment (Kojoley) [inline scripts/boost/generate-ports.ps1:373]: Boost 1.77 does not contain dependency cycles.

Comment (yurybura) [reply scripts/boost/generate-ports.ps1]: > Boost 1.77 does not contain dependency cycles.

Script has been fixed. Thank you.

File: scripts/boost/generate-ports.ps1 [hunk 2]
```diff
@@ -223,214 +222,236 @@ function Generate()
...
     pushd $unpacked
-    try
-    {
-        $groups = Get-ChildItem -Recurse -Path include,src -File `
+    try {
+        $groups = Get-ChildItem -Recurse -Path include, src -File `
             | ? { $_ -is [System.IO.FileInfo] } `
             | % { Get-Content -LiteralPath $_ } `
-            | ? { $_ -match 'include [<"]boost/' }
+            | ? { $_ -match '.*# *include [<"]boost\/' }
```
Comment (Kojoley) [inline scripts/boost/generate-ports.ps1:298]: Multiple spaces are allowed between `include` token and the path, for some reason in different places of this it is handled or not.

Comment (yurybura) [reply scripts/boost/generate-ports.ps1:298]: @Kojoley Thank you for comment!
Multiple spaces and other not interesting symbols will be removed at line 321:
```
# extract path
$_ -replace " *# *include *[<`"]boost\/([a-zA-Z0-9\.\-_\/]*)[>`"].*", "`$1"
```
After this the string should contain only path.

Comment (yurybura) [reply scripts/boost/generate-ports.ps1]: Should be fixed in c7ca2c7

**Comment by yurybura** [comment]
@Kojoley 
Dear Nikita, could you help with B2 issue (already reported in https://github.com/boostorg/build/issues/724)?

Error log:
```
D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\targets.jam:609: in start-building from module targets
error: Recursion in main target references
error: the following target are being built currently:
error: ../build/boost_nowide /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_defaulted_functions /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_noexcept /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_rvalue_references /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_static_assert /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/config/cxx11_moveable_fstreams /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/config/lfs_support /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/config/attribute_init_priority /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_defaulted_functions
D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\targets.jam:776: in class@main-target.generate from module object(main-target)@1829
D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\targets.jam:934: in targets.generate-from-reference from module targets
D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\configure.jam:355: in builds-raw from module configure
...
```

**Comment by Kojoley** [comment]
> @Kojoley
> Dear Nikita, could you help with B2 issue (already reported in [boostorg/build#724](https://github.com/boostorg/build/issues/724))?
> 
> Error log:
> 
> ```
> D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\targets.jam:609: in start-building from module targets
> error: Recursion in main target references
> error: the following target are being built currently:
> error: ../build/boost_nowide /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_defaulted_functions /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_noexcept /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_rvalue_references /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_static_assert /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/config/cxx11_moveable_fstreams /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/config/lfs_support /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/config/attribute_init_priority /D:/libs/yurybura/vcpkg/buildtrees/boost-nowide/src/ost-1.77.0-2944b29e72.clean/build/config/checks/cxx11_defaulted_functions
> D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\targets.jam:776: in class@main-target.generate from module object(main-target)@1829
> D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\targets.jam:934: in targets.generate-from-reference from module targets
> D:/libs/yurybura/vcpkg/installed/x64-windows/tools/boost-build/src/build\configure.jam:355: in builds-raw from module configure
> ...
> ```

I do not think it is a B2 issue. I bet the problem is in the destination path of this: https://github.com/microsoft/vcpkg/blob/63aa65e65b9d2c08772ea15d25fb8fdb0d32e557/scripts/boost/post-source-stubs/nowide.cmake#L6

B2 project requirements are applied to every subproject, and vcpkg puts Config code inside Nowide what creates a recursive cycle -- Config requirements get Nowide requirements what leads to Config targets require Config targets.

Ideally, vcpkg should just remove the hack. Was the use case lead to this hack reported to the Boost dev mail list to find a better solution? Even if it was, it would not hurt to remind that the problem still exists.

**Comment by yurybura** [comment]
@Kojoley Thank you for the answer. I agree with you, that this issue is related to copying of configs. But before v1.76.0 everything worked fine. The main problem here is that `build/Jamfile.v2` imports checks from non exist relative path:
```
import ../../config/checks/config : requires ;
```
Let me think how to fix this problem without copying.

**Comment by Kojoley** [comment]
> @Kojoley Thank you for the answer. I agree with you, that this issue is related to copying of configs. But before v1.76.0 everything worked fine.

Working before does not imply vcpkg was not doing something wrong, it means that B2 did not detect the issue before.

> The main problem here is that `build/Jamfile.v2` imports checks from non exist relative path:
> 
> ```
> import ../../config/checks/config : requires ;
> ```

Boost become modular but its libraries are still not standalone, they expect to be a part of Boost.

> Let me think how to fix this problem without copying.

I have no idea how vcpkg is building Boost, so I do not know.

**Comment by autoantwort** [comment]
Maybe we can simply remove the checks or use the boost build version 1.75.

**Comment by yurybura** [comment]
> Maybe we can simply remove the checks or use the boost build version 1.75.
Using old version of Boost.Build causes another issue related to PCH (precompile headers).

File: ports/boost-json/001-remove-checks.patch [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ #
+ 
+-import ../../config/checks/config : requires ;
+-
+ project boost/json
+     : requirements
+       $(c11-requires)
+       <link>shared:<define>BOOST_JSON_DYN_LINK=1
+       <link>static:<define>BOOST_JSON_STATIC_LINK=1
+-      <library>/boost//container/<warnings-as-errors>off
```
[the remark this answers is not in the document]
Comment (yurybura) [reply ports/boost-json/001-remove-checks.patch]: Fixed.

**Comment by yurybura** [comment]
@PhoebeHui @ras0219-msft @BillyONeal Please start review. Thanks!

**Comment by PhoebeHui** [comment]
I have double checked:

- [ ] boost-regex[icu] with x64-windows and x64-linux test pass
- [ ] boost-locale[icu] with x64-windows and x64-linux test pass
- [ ] wt: all  features with x64-window test pass
- [ ] Boost-fiber/boost-thread test pass when build with release only, which used to test the changes in ports/boost-modular-build-helper/Jamroot.jam.
- [ ] The version of boost-build/boost-uninstall/boost-vcpkg-helpers has been updated
- [ ] Scripts/boost/generate-ports.ps1, the changes need other members further review.

**Comment by PhoebeHui** [comment]
cc @cenit @autoantwort

**Comment by autoantwort** [comment]
Why was qwtw added to the baseline?

**Comment by yurybura** [comment]
> Why was qwtw added to the baseline?

This lib has a build problem. I have no idea how to fix this. I cannot update this library because newer releases of this project have lower versions numbers.. As a result, I reported this issue in the original repository https://github.com/ig-or/qwtw/issues/2 and added it to the baseline.

**Comment by cenit** [comment]
is this working in single-config triplets? in #15983 i had to do some fixes to boost to make it work in those triplets....

**Comment by yurybura** [comment]
> is this working in single-config triplets? in #15983 i had to do some fixes to boost to make it work in those triplets....

I haven't tested this config. I think it may work. Let me know steps to test it.

**Review by autoantwort — APPROVED** [review]

**Comment by cenit** [comment]
> > is this working in single-config triplets? in #15983 i had to do some fixes to boost to make it work in those triplets....
> 
> I haven't tested this config. I think it may work. Let me know steps to test it.

Add 

```
set(VCPKG_BUILD_TYPE “release”)
```

to a triplet (a new one or an existing one) and then build boost with that triplet. Please also see the above PR to see what i had to modify to the jam build script

**Comment by yurybura** [comment]
> > > is this working in single-config triplets? in #15983 i had to do some fixes to boost to make it work in those triplets....
> > 
> > 
> > I haven't tested this config. I think it may work. Let me know steps to test it.
> 
> Add
> 
> ```
> set(VCPKG_BUILD_TYPE “release”)
> ```
> 
> to a triplet (a new one or an existing one) and then build boost with that triplet. Please also see the above PR to see what i had to modify to the jam build script

I've just tested custom triplet `x64-windows-release`. Boost works like expected.

**Comment by PhoebeHui** [comment]
Yes, I have tested those changes with x64-linux-release, all built successfully.

**Review by PhoebeHui — APPROVED** [review]
LGTM

**Comment by sketch34** [comment]
Thanks guys, just to chip in I've tested this PR and it works fine for me with `boost::asio` and `boost::beast` @ 1.77.0 👍🏼

**Comment by BillyONeal** [comment]
Thanks for the updated copy of Boost!

**Comment by sketch34** [comment]
Thanks to everyone and @yurybura for the PR 👏

**Pull request merged**